### PR TITLE
KPO write_logs does not need to use complicated pod logs reader

### PR DESCRIPTION
**Pull request by dstandish**
Method `write_logs` was using pod manager method `read_pod_logs` which has some rather complicated logic that it doesn't need, because it does not "follow" the logs and because by the time we come out of deferral, the pod is already done.

By reducing usage of `read_pod_logs`, it makes it easier to hopefully refactor and simplify some of our KPO log fetching logic.

I also make the metheod explicitly private which it should have been anyway.

### CODE REVIEW

File: airflow/providers/cncf/kubernetes/operators/pod.py [hunk 1]
```diff
@@ -703,22 +703,21 @@ def execute_complete(self, context: Context, event: dict, **kwargs):
                     remote_pod=pod,
                 )
 
-    def write_logs(self, pod: k8s.V1Pod):
+    def _write_logs(self, pod: k8s.V1Pod):
```
Comment (eladkal) [inline airflow/providers/cncf/kubernetes/operators/pod.py:654]: is this considered a breaking change?

Comment (dstandish) [reply airflow/providers/cncf/kubernetes/operators/pod.py:654]: I don't think we need to consider it breaking.  This is a method that realistically no one would ever override.  It's used in one place.  I think of it as "bugfix" that it wasn't marked private.

Really this also connects to something we've discussed before.  That our operators should not be published as "base classes" to be extended.  I think they should be end products whose behavior is governed by semver but not whose _structure_ is governed by semver.  Personally I think we ought to update our "public interface" doc to reflect this.

Comment (eladkal) [reply airflow/providers/cncf/kubernetes/operators/pod.py:654]: cool yeah I see the discussion in https://github.com/apache/airflow/pull/36767

**Review by eladkal — APPROVED** [review]

File: airflow/providers/cncf/kubernetes/operators/pod.py (hunk 1 above)
Comment (amoghrajesh) [reply airflow/providers/cncf/kubernetes/operators/pod.py:654]: Checked the discussion in https://github.com/apache/airflow/pull/36767, seems about right to me that no one "should" ever override the write_logs method, it is something that should not be changed. Seems like a non breaking change

File: airflow/providers/cncf/kubernetes/operators/pod.py [hunk 2]
```diff
@@ -651,22 +651,21 @@ def execute_complete(self, context: Context, event: dict, **kwargs):
...
                 line = raw_line.decode("utf-8", errors="backslashreplace").rstrip("\n")
-                self.log.info("Container logs: %s", line)
+                self.log.info("[%s] %s", self.base_container_name, line)
         except HTTPError as e:
-            self.log.warning(
-                "Reading of logs interrupted with error %r; will retry. "
-                "Set log level to DEBUG for traceback.",
-                e,
-            )
+            self.log.warning("Reading of logs interrupted with error %r;", e)
```
Comment (amoghrajesh) [inline airflow/providers/cncf/kubernetes/operators/pod.py:668]: Does it retry? We should mention in the warning in that case.

**Review by amoghrajesh — APPROVED** [review]
Looks good, one nit/ask, rest is fine 👍🏽

**Pull request closed without merging**